src/Files.App/Views/LayoutModes/ColumnViewBase.xaml.cs

```diff
@@ -575,18 +555,23 @@ private void FileList_ItemTapped(object sender, TappedRoutedEventArgs e)
 				{
 					CheckRenameDoubleClick(clickedItem.DataContext);
 				}
-				else if (IsRenamingItem)
+				else if (IsRenamingItem &&
+					FileList.ContainerFromItem(RenamingItem) is ListViewItem listViewItem &&
+					listViewItem.FindDescendant("ListViewTextBoxItemName") is TextBox textBox)
+				{
+					CommitRename(textBox);
+				}
+
+				if (isItemFolder && UserSettingsService.FoldersSettingsService.ColumnLayoutOpenFoldersWithOneClick)
 				{
-					if (FileList.ContainerFromItem(RenamingItem) is ListViewItem listViewItem
-						&& listViewItem.FindDescendant("ListViewTextBoxItemName") is TextBox textBox)
-					{
-						CommitRename(textBox);
-					}
+					ItemInvoked?.Invoke(new ColumnParam { NavPathParam = (item is ShortcutItem sht ? sht.TargetPath : item!.ItemPath), ListView = FileList }, EventArgs.Empty);
```

Comment:
I'd suggest spacing the ColumnParam on several lines to make it easier to read. 
```suggestion
					ItemInvoked?.Invoke(
						new ColumnParam 
						{ 
							NavPathParam = (item is ShortcutItem sht ? sht.TargetPath : item!.ItemPath), 
							ListView = FileList 
						}, 
						EventArgs.Empty);
```